File: leetCode/array/Remove Element.py
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        i = 0
        while True:
          if i >= len(nums):
            break
          if nums[i] == val:
            nums.pop(i)
          else:
            i += 1
        return len(nums)

test = Solution()
print(test.removeElement([3,2,2,3], 3))

File: leetCode/array/Remove Element.js
/**
 * @param {number[]} nums
 * @param {number} val
 * @return {number}
 */
var removeElement = function(nums, val) {
    let left = 0;
    for (let num of nums ) {
        if (num !== val) {
            nums[left++] = num;
        }
    }
    return left;
    
};